### PR TITLE
Add essence-feature-usage-stats tool

### PR DESCRIPTION
**Pull request by gskorokhod**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,10 @@
 target
 
 solvers/**/vendor/build
+tools/essence-feature-usage-stats/.venv
```
Comment (ozgurakgun) [inline .gitignore]: .venv
__pycache__
.idea

**Comment by ozgurakgun** [comment]
hi @gskorokhod - let me know when you add the github action please.

**Comment by ozgurakgun** [comment]
I suspect we don't need the generated html in this branch, just on the gh-pages branch (after deployment).

**Comment by niklasdewally** [comment]
> I suspect we don't need the generated html in this branch, just on the gh-pages branch (after deployment).

Georgii and I were discussing how to put stuff to gh pages yesterday (primarily uploading artifacts vs hosting the entire docs folder)

To clarify, are we treating pages as its own branch, or as the doc/ folder. (I know GH supports both?)

**Comment by ozgurakgun** [comment]
A separate branch. I had sent Georgii an example from another repository where we used a GitHub action to deploy to gh-pages on a branch. I can find this again if he doesn't have it.

**Comment by ozgurakgun** [comment]
Found it: https://github.com/conjure-cp/task-allocation-demo/blob/14b930ff29e6b25724df01e1cc47a1d8ead04d57/.github/workflows/deploy.yml

**Comment by gskorokhod** [comment]
@ozgurakgun I've finished setting up the action - I think it is ready to review?

**Comment by niklasdewally** [comment]
(georgii has a lecture, so I am pushing a commit or two of cleanup)

**Comment by gskorokhod** [comment]
Also made columns hideable, made it possible to require/exclude specific keywords, and made the table look slightly nicer

File: tools/essence-feature-usage-stats/.env [hunk 2]
```diff
@@ -0,0 +1,11 @@
+ESSENCE_DIR=/home/mayday/Coding/VIP/EssenceCatalog
```
Comment (ozgurakgun) [inline tools/essence-feature-usage-stats/.env:1]: are these local paths still needed?

File: tools/essence-feature-usage-stats/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+black==23.10.1
```
Comment (ozgurakgun) [inline tools/essence-feature-usage-stats/requirements.txt:1]: we should ask dependabot to monitor these as well

**Review by ozgurakgun — COMMENTED** [review]
couple of minor comments, almost there!

File: tools/essence-feature-usage-stats/.env (hunk 2 above)
Comment (gskorokhod) [reply tools/essence-feature-usage-stats/.env:1]: No, the whole .env file is just for local testing. All the necessary env vars are in the GitHub Action

Comment (ozgurakgun) [reply tools/essence-feature-usage-stats/.env:1]: Should we remove them from the repo then? As they are, they won't work on anybody else's computer.

Comment (gskorokhod) [reply tools/essence-feature-usage-stats/.env:1]: True! I have removed the file, just waiting for the CI to complete and the commit will appear here

**Comment by gskorokhod** [comment]
Removed the .env and updated dependabot.yml file (hopefully it should track the Python packages now?)

File: tools/essence-feature-usage-stats/requirements.txt (hunk 3 above)
Comment (gskorokhod) [reply tools/essence-feature-usage-stats/requirements.txt:1]: done

**Comment by ozgurakgun** [comment]
@gskorokhod - happy for me to merge this?

**Comment by gskorokhod** [comment]
> @gskorokhod - happy for me to merge this?

Yes, I think it’s done!

**Pull request merged**